zephyr/include/rtos/interrupt.h

```diff
@@ -83,10 +83,24 @@ static inline uint32_t interrupt_disable(uint32_t irq, void *arg)
 	return 0;
 }
 
+#if CONFIG_ARM64
+/* these are only required for linkage purposes.
+ *
+ * in the case of arm64 the interrupts don't
+ * need to be ACK'd explicitly as it's already
+ * done after the ISR handler is executed. Also
+ * said interrupts are never masked so there's
+ * also no need for interrupt_clear_mask().
+ */
+#define interrupt_clear(irq)
+#define interrupt_clear_mask(irq, mask)
+#else
+void interrupt_clear_mask(uint32_t irq, uint32_t mask);
+#endif /* CONFIG_ARM64 */
+
```

Comment:
And this is really only needed with XTOS. With native Zephyr SOF build, the interrupt interface for drivers is all on Zephyr side. SOF app side only needs primitives to disable interrupts, but that can be done with generic OS interface.

But alas, if you want to maintain XTOS support with ARM64, then you need this.

Reply:
This _looks_ like there's not going to be any XTOS support on ARM64, but correct me if I'm wrong Laurentiu.

We do want to eventually phase out XTOS for the older i.MX platforms as well, but that's not happening yet.

Reply:
This was required because of some compilation errors caused by these symbols not existing (e.g: for zephyr_dma_domain which is also used for Zephyr+XTOS).

Reply:
Aforementioned issue should be fixed by https://github.com/thesofproject/sof/pull/7470 and  https://github.com/thesofproject/sof/pull/7462. Although not necessarily related to i.MX93 driver support I've added the last commit over there to avoid adding an extra PR dependency as it's already somewhat hard to keep track of them.